enginetest/enginetests.go

```diff
@@ -5502,217 +5480,12 @@ func TestAlterTable(t *testing.T, harness Harness) {
-		RunQuery(t, e, harness, "ALTER TABLE t39 DROP INDEX u_col1_col2;")
-		RunQuery(t, e, harness, "INSERT INTO t39 VALUES (10, 1, 1);")
-		AssertErr(t, e, harness, "ALTER TABLE t39 ADD UNIQUE u_col1_col2 (col1, col2)", sql.ErrUniqueKeyViolation)
-		tt = queries.QueryTest{
-			Query: "show create table t39;",
-			Expected: []sql.Row{{"t39", "CREATE TABLE `t39` (\n" +
-				"  `pk` int NOT NULL,\n" +
-				"  `col1` int,\n" +
-				"  `col2` int,\n" +
-				"  PRIMARY KEY (`pk`)\n" +
-				") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin"}},
-		}
-		TestQueryWithEngine(t, harness, e, tt)
-	})
-
-	t.Run("ALTER TABLE remove AUTO_INCREMENT", func(t *testing.T) {
-		RunQuery(t, e, harness, "CREATE TABLE t40 (pk int AUTO_INCREMENT PRIMARY KEY, val int)")
-		RunQuery(t, e, harness, "INSERT into t40 VALUES (1, 1), (NULL, 2), (NULL, 3)")
-
-		RunQuery(t, e, harness, "ALTER TABLE t40 MODIFY COLUMN pk int")
-		ctx := harness.NewContext()
-		TestQueryWithContext(t, ctx, e, harness, "DESCRIBE t40", []sql.Row{
-			{"pk", "int", "NO", "PRI", "NULL", ""},
-			{"val", "int", "YES", "", "NULL", ""}},
-			nil, nil)
-
-		AssertErr(t, e, harness, "INSERT INTO t40 VALUES (NULL, 4)", sql.ErrInsertIntoNonNullableProvidedNull)
-		RunQuery(t, e, harness, "DROP TABLE t40")
-
-		RunQuery(t, e, harness, "CREATE TABLE t40 (pk int AUTO_INCREMENT PRIMARY KEY, val int)")
-		RunQuery(t, e, harness, "INSERT into t40 VALUES (NULL, 1)")
-
-		TestQueryWithContext(t, ctx, e, harness, "SELECT * FROM t40", []sql.Row{{1, 1}}, nil, nil)
-	})
-
-	TestScript(t, harness, queries.ScriptTest{
-		// https://github.com/dolthub/dolt/issues/6206
-		Name: "alter table containing column default value expressions",
-		SetUpScript: []string{
-			"create table t (pk int primary key, col1 timestamp default current_timestamp(), col2 varchar(1000), index idx1 (pk, col1));",
-		},
-		Assertions: []queries.ScriptTestAssertion{
-			{
-				Query:    "alter table t alter column col2 DROP DEFAULT;",
-				Expected: []sql.Row{},
-			},
-			{
-				Query:    "show create table t;",
-				Expected: []sql.Row{{"t", "CREATE TABLE `t` (\n  `pk` int NOT NULL,\n  `col1` timestamp(6) DEFAULT (CURRENT_TIMESTAMP()),\n  `col2` varchar(1000),\n  PRIMARY KEY (`pk`),\n  KEY `idx1` (`pk`,`col1`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin"}},
-			},
-			{
-				Query:    "alter table t alter column col2 SET DEFAULT 'FOO!';",
-				Expected: []sql.Row{},
-			},
-			{
-				Query:    "show create table t;",
-				Expected: []sql.Row{{"t", "CREATE TABLE `t` (\n  `pk` int NOT NULL,\n  `col1` timestamp(6) DEFAULT (CURRENT_TIMESTAMP()),\n  `col2` varchar(1000) DEFAULT 'FOO!',\n  PRIMARY KEY (`pk`),\n  KEY `idx1` (`pk`,`col1`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin"}},
-			},
-			{
-				Query:    "alter table t drop index idx1;",
-				Expected: []sql.Row{{types.NewOkResult(0)}},
-			},
-		},
-	})
+func TestAlterTableScripts(t *testing.T, harness Harness) {
```

Comment:
Any reason not to just run these in the existing `TestAlterTable` function? Would be one less test function to wire through in dolt_engine_tests and memory_engine_tests and seems like they make sense together.

Reply:
Makes sense